File: src/modules/commands/dog.rb
require 'httparty'
require 'json'

module Bot::DiscordCommands
  module Dog
    extend Discordrb::Commands::CommandContainer
    command([:dog, :doggo],
            description: '',
            usage: '') do |event|
      response = HTTParty.get('https://random.dog/woof.json')
      fact = HTTParty.get('https://some-random-api.ml/facts/dog')

      event.channel.send_embed do |embed|
        embed.title = ['🐶', '🐕', '🐩'].sample
        embed.url = JSON.parse(response.body)['url']

        embed.description = "Cool Dog Fact #{rand(300)}: #{JSON.parse(fact.body)['fact']}"

        embed.image = Discordrb::Webhooks::EmbedImage.new(url: JSON.parse(response.body)['url'])

        embed.color = Bot::CONFIG.color
        embed.timestamp = Time.now
        embed.footer = Discordrb::Webhooks::EmbedFooter.new(text: "Requested by #{event.message.author.username}", icon_url: event.message.author.avatar_url)
      end
    end
  end
end

File: src/commands/dog.js
const { Command } = require('discord-akairo')
const { embedColor } = require('../../data/config')
const got = require('got')

class DogCommand extends Command {
  constructor () {
    super('dog', {
      aliases: [ 'dog', 'doggo' ]
    })
  }

  async exec (message) {
    const embed = this.client.util.embed()
    const dogs = [ '🐶', '🐕', '🐩' ]

    const image = await got('https://random.dog/woof.json', { json: true })
    const fact = await got('https://some-random-api.ml/facts/dog', { json: true })

    embed.setTitle(dogs[Math.floor(Math.random() * dogs.length)])
    embed.setURL(image.body.url)
    embed.setDescription(fact.body.fact)

    embed.setImage(image.body.url)

    embed.setFooter(`Requested by ${message.author.username}#${message.author.discriminator}`, message.author.avatarURL)
    embed.setTimestamp(new Date())
    embed.setColor(embedColor)

    await message.channel.send(embed)
  }
}

module.exports = DogCommand
